File: Labs/Lab 1/C++/StringHandler.cpp
#include "StringHandler.h"
#include <algorithm>

using namespace std;

// Constructor initializes the string
StringHandler::StringHandler(const string &str) : data(str) {}

// Returns the length of the string
int StringHandler::getLength() const { return data.length(); }

// Returns the string sorted in descending order
string StringHandler::getSortedDescending() const {
  string sortedStr = data;
  sort(sortedStr.rbegin(), sortedStr.rend()); // Sort in descending order
  return sortedStr;
}

// Returns the original string
string StringHandler::getValue() const { return data; }

File: Labs/Lab 1/С#/StringHandler.cs
// filepath: /c:/Users/user/Documents/Programming-Fundamentals-OOP/Labs/Lab 1/С#/StringHandlerApp/StringHandler.cs
using System;
using System.Linq;

namespace StringHandlerApp
{
    public class StringHandler
    {
        private string data; // Stores the string data

        public StringHandler(string str)
        {
            data = str; // Initialize string
        }

        public int GetLength()
        {
            return data.Length; // Returns the length of the string
        }

        public string GetSortedDescending()
        {
            return new string(data.OrderByDescending(c => c).ToArray()); // Sort characters in descending order
        }

        public string GetValue()
        {
            return data; // Returns the original string
        }
    }
}